tests/python/test_fields_builder.py

```diff
@@ -3,174 +3,276 @@
 
 import taichi as ti
 
+'''
+Test fields with shape.
+'''
+
 
 @ti.test(arch=[ti.cpu, ti.cuda, ti.vulkan, ti.metal])
 def test_fields_with_shape():
-    n = 5
-    x = ti.field(ti.f32, [n])
 
+    # Initialize common variables for tests below
+    shape_size_1d = 5
+    x = ti.field(ti.f32, [shape_size_1d])
+    y = ti.field(ti.f32, [shape_size_1d])
+
+    # [shape] 1. Test and validation for assign once kernel function
     @ti.kernel
-    def func():
-        for i in range(n):
+    def assign_once_kernel_func():
```

Comment:
maybe just `assign_field`? `*_kernel_func` is a bit confusing.

Reply:
same for all occurrences below

Reply:
fixed